org.scala-ide.sdt.core/src/org/scalaide/core/completion/ProposalRelevanceCfg.scala

```diff
@@ -29,3 +35,22 @@ object DefaultProposalRelevanceCfg extends ProposalRelevanceCfg {
     """play\.(?!api\.).*""".r ::
     Nil
 }
+
+object PrefsBasedProposalRelevanceCfg extends ProposalRelevanceCfg {
+  val PFavoritePackages = "scala.tools.eclipse.ui.preferences.completions.favoritePackages"
```

Comment:
Please move everything that has to do with the pref store and all of these constants to `CompletionPreferencePage` because that is the location where we usually keep this stuff.